From d5370337585b7a812ddf25e0280c2643c9aa9409 Mon Sep 17 00:00:00 2001
From: Hai Zhang <zhanghai@google.com>
Date: Mon, 31 Aug 2020 18:28:41 +0000
Subject: [PATCH 237/297] DO NOT MERGE Fix NPE in executeDeletePackageLIF.

The parameter allUserHandles may sometimes be null.

Change-Id: Ie58cabe4bb1cae7173ad1d027e993a0b82a9f1ef
Bug: 167233572
Bug: 140256621
Test: atest EphemeralTest
(cherry picked from commit 01e0f4813af8502faaa683d55dbbc024b3080741)
---
 .../core/java/com/android/server/pm/PackageManagerService.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 81c111fc364..6496cd31031 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -19281,7 +19281,8 @@ public class PackageManagerService extends IPackageManager.Stub
 
         // We need to get the permission state before package state is (potentially) destroyed.
         final SparseBooleanArray hadSuspendAppsPermission = new SparseBooleanArray();
-        for (int userId : allUserHandles) {
+        // allUserHandles could be null, so call mUserManager.getUserIds() directly which is cached anyway.
+        for (int userId : mUserManager.getUserIds()) {
             hadSuspendAppsPermission.put(userId, checkPermission(Manifest.permission.SUSPEND_APPS,
                     packageName, userId) == PERMISSION_GRANTED);
         }
-- 
2.36.0

